common/repositories/character_data_repository.h

```diff
@@ -64,6 +64,134 @@ class CharacterDataRepository: public BaseCharacterDataRepository {
 
 		return Strings::ToUnsignedInt(row[0]);
 	}
+
+	static CharacterData FindByName(
+		Database& db,
+		const std::string& character_name
+	)
+	{
+		auto results = db.QueryDatabase(
+			fmt::format(
+				"{} WHERE `name` = '{}' LIMIT 1",
+				BaseSelect(),
+				Strings::Escape(character_name)
+			)
+		);
+
+		auto row = results.begin();
+		if (results.RowCount() == 1) {
+			CharacterData e{};
+
+			e.id                      = row[0] ? static_cast<uint32_t>(strtoul(row[0], nullptr, 10)) : 0;
+			e.account_id              = row[1] ? static_cast<int32_t>(atoi(row[1])) : 0;
+			e.name                    = row[2] ? row[2] : "";
+			e.last_name               = row[3] ? row[3] : "";
+			e.title                   = row[4] ? row[4] : "";
+			e.suffix                  = row[5] ? row[5] : "";
+			e.zone_id                 = row[6] ? static_cast<uint32_t>(strtoul(row[6], nullptr, 10)) : 0;
+			e.zone_instance           = row[7] ? static_cast<uint32_t>(strtoul(row[7], nullptr, 10)) : 0;
+			e.y                       = row[8] ? strtof(row[8], nullptr) : 0;
+			e.x                       = row[9] ? strtof(row[9], nullptr) : 0;
+			e.z                       = row[10] ? strtof(row[10], nullptr) : 0;
+			e.heading                 = row[11] ? strtof(row[11], nullptr) : 0;
+			e.gender                  = row[12] ? static_cast<uint8_t>(strtoul(row[12], nullptr, 10)) : 0;
+			e.race                    = row[13] ? static_cast<uint16_t>(strtoul(row[13], nullptr, 10)) : 0;
+			e.class_                  = row[14] ? static_cast<uint8_t>(strtoul(row[14], nullptr, 10)) : 0;
+			e.level                   = row[15] ? static_cast<uint32_t>(strtoul(row[15], nullptr, 10)) : 0;
```

Comment:
We don't want to be defining all of this manually and should be leaving it to code generation.

We'd be better off having a `GetWhere` here, return a vector/array of struct entries and just return the first entry and remove all of this